src/commands/env/create/sandbox.ts

```diff
@@ -0,0 +1,309 @@
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-env', 'create.sandbox');
+const progressBarOptions = {
+  title: 'Sandbox Create',
+  format: '%s | {id} | {status} | {bar} | {percentComplete}%. {remainingWaitTimeHuman}',
+  barCompleteChar: '\u2588',
+  barIncompleteChar: '\u2591',
+  linewrap: true,
+};
+
+export enum SandboxLicenseType {
+  Developer = 'Developer',
+  DeveloperPro = 'Developer_Pro',
+  Partial = 'Partial',
+  Full = 'Full',
+}
+
+const getLicenseTypes = (): string[] => Object.values(SandboxLicenseType);
+
+export default class CreateSandbox extends SfCommand<SandboxProcessObject> {
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public static flags: Interfaces.FlagInput<any> = {
+    // needs to change when new flags are available
+    'definition-file': Flags.file({
+      exists: true,
+      char: 'f',
+      summary: messages.getMessage('flags.definitionFile.summary'),
+      exclusive: ['name', 'license-type'],
+    }),
+    'set-default': Flags.boolean({
+      char: 's',
+      summary: messages.getMessage('flags.setDefault.summary'),
+    }),
+    alias: Flags.string({
+      char: 'a',
+      summary: messages.getMessage('flags.alias.summary'),
+    }),
+    wait: Flags.duration({
+      char: 'w',
+      summary: messages.getMessage('flags.wait.summary'),
+      min: 6,
+      unit: 'minutes',
+      defaultValue: 6,
+    }),
+    name: Flags.string({
+      char: 'n',
+      summary: messages.getMessage('flags.name.summary'),
+      exclusive: ['definition-file'],
+    }),
+    'license-type': Flags.enum({
+      char: 'l',
+      summary: messages.getMessage('flags.licenseType.summary'),
+      exclusive: ['definition-file'],
+      options: getLicenseTypes(),
+      default: SandboxLicenseType.Developer,
+    }),
+    'target-org': Flags.requiredOrg({
+      char: 'o',
+      summary: messages.getMessage('flags.targetOrg.summary'),
+    }),
+    'no-prompt': Flags.boolean({
+      summary: messages.getMessage('flags.noPrompt.summary'),
+    }),
+  };
+
+  protected readonly lifecycleEventNames = ['postorgcreate'];
+  private sandboxAuth?: SandboxUserAuthResponse;
+  private flags: {
+    'definition-file': string;
+    'set-default': boolean;
+    alias: string;
+    wait: Duration;
+    json: boolean;
+    name: string;
+    'license-type': SandboxLicenseType;
+    'no-prompt': boolean;
+    'target-org': Org;
+  };
+
+  public async run(): Promise<SandboxProcessObject> {
+    this.flags = (await this.parse(CreateSandbox)).flags as CreateSandbox['flags'];
+    this.debug('Create started with args %s ', this.flags);
+
+    this.validateSandboxFlags();
+    return await this.createSandbox();
+  }
+
+  private validateSandboxFlags(): void {
```

Comment:
Not following

Reply:
rather than loading/parsing all the flags, then validating the values of those in a separate private method, each flag that need validation *could* use its `parse` function to do validation on its input (if any is provided)

That way, the first flag that fails (ex: an invalid sandbox name) stops the processing early?

Reply:
Thanks, I'll have a look.